libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1147,6 +1146,62 @@ string YulUtilFunctions::clearStorageArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::clearStorageStructFunction(StructType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+
+	string functionName = "clear_struct_storage_" + _type.identifier();
+
+	return m_functionCollector.createFunction(functionName, [&] {
+		MemberList::MemberMap structMembers = _type.nativeMembers(nullptr);
+		vector<map<string, string>> memberSetValues;
+
+		set<u256> slotsCleared;
+		for (auto const& member: structMembers)
+			if (member.type->storageBytes() < 32)
```

Comment:
Can you extract this part into its own PR?

Reply:
You mean the whole function?

Reply:
The first optimizing case.

Reply:
I removed it from here. Will open another PR with optimization.

Reply:
#9985 is where I extracted optimizing case. Also it targets this branch as it depends on this one.